index.js

```diff
@@ -53,14 +53,24 @@ function ensureAllCommentsPrinted(astComments) {
   });
 }
 
-function format(text, opts, addAlignmentSize) {
+function formatWithCursor(text, opts, addAlignmentSize) {
   addAlignmentSize = addAlignmentSize || 0;
 
   const ast = parser.parse(text, opts);
 
   const formattedRangeOnly = formatRange(text, opts, ast);
   if (formattedRangeOnly) {
-    return formattedRangeOnly;
+    return { formatted: formattedRangeOnly };
+  }
```

Comment:
The cursor-translation functionality still needs to be unified with the range-formatting functionality (currently, either one or the other can be used). I think it might be easiest to do this by keeping track of the range start/end positions in the same way as the cursor positions, then having an API function that returns all 3 positions (if provided), and only formats the text necessary to satisfy the range.